pkg/apis/eventing/v1/trigger_validation.go

```diff
@@ -163,3 +146,72 @@ func (t *Trigger) validateInjectionAnnotation(injectionAnnotation string) *apis.
 	}
 	return nil
 }
+
+func ValidateAttributeFilters(filter *TriggerFilter) (errs *apis.FieldError) {
+	if filter == nil {
+		return nil
+	}
+	return errs.Also(ValidateAttributesNames(filter.Attributes).ViaField("attributes"))
+}
+
+func ValidateAttributesNames(attrs map[string]string) (errs *apis.FieldError) {
+	if attrs == nil {
+		return nil
+	}
```

Comment:
nit: Can be removed. It is redundant because if `attrs` is nil, the `for` loop will be a no-op and `errs` will effectively be nil too.

Also, in other validation functions the comparison is `len(x) == 0` (which is recommended because you could get a non-nil empty map), so even if you choose not to delete this conditional for \<reason>, it would be sensible to compare the `len` instead of `== nil` here too.